noir_stdlib/src/hash/sha256.nr

```diff
@@ -114,13 +127,13 @@ pub fn sha256_var<let N: u32>(msg: [u8; N], message_size: u64) -> HASH {
     // Pad the rest such that we have a [u32; 2] block at the end representing the length
     // of the message, and a block of 1 0 ... 0 following the message (i.e. [1 << 7, 0, ..., 0]).
     // Here we rely on the fact that everything beyond the available input is set to 0.
-    msg_block[msg_byte_ptr] = 1 << 7;
-    let last_block = msg_block;
+    msg_block = set_msg_byte(msg_block, msg_byte_ptr, 1 << 7);
```

Comment:
This (implementation of `set_msg_byte`) feels like it can be simplified as we know that the the contents of `msg_block` after where we're inserting the byte is all zeros. We then don't need to extra just that single byte but can pull out everything after it as well.

Reply:
I thought this is one of the most straight forward ways to implement it, but you're right in this instance it might not be the most optimal. Bit of a brain teaser every time I have to touch one of these items, so I thought simpler is good. I'll have a look at a more specialised approach.

Reply:
Added a bunch of new helpers and now setting not just a specific byte in the number but all following bytes to zero as well. This saves us the work of reading what was in that position, because we don't have to preserve the others by subtracting it.